### DISCUSSION TITLE
Publicly host older tendermint state/block data

### DISCUSSION THREAD

**Discussion by Esya** [oasisprotocol/community-forum#20] [Ideas]
Cross-linking with https://github.com/oasisprotocol/oasis-core/issues/3878 : 

> Currently, once an upgrade comes in (like v21.X) and a new genesis file is generated and used by the nodes, the previous state is dismissed and the genesis contains the ledger of all active accounts.

Once we migrate to 21.X the states of the nodes will be wiped, and new nodes joining the network will only pull blocks from the new start height of the genesis. While this helps maintaining the state light on nodes, this is an issue both for transparency/auditability of the chain, but also for community-written tools like block explorers that will lose access to historical data.

In https://github.com/oasisprotocol/oasis-core/issues/3878 I propose we start discussing a way to offer the same kind of APIs through oasis-core for historical data, but in the mean time I think there is a discussion to be had around publicly hosting and sharing historical states so that it is not just lost on the mainnet launch to cobalt upgrade period.

**Comment by kostko** [comment]
> I think there is a discussion to be had around publicly hosting and sharing historical states so that it is not just lost on the mainnet launch to cobalt upgrade period.

Agreed, this sounds like a good idea. The Foundation is running a full node which is keeping all state throughout history and so after the upgrade we could set up a public repository somewhere and upload the state snapshot there, signed by some of our GPG keys. Others running archival nodes (e.g. block explorers) may opt to do something similar. Thoughts?

**Reply by Esya** [reply]
I think this definitely makes sense as something that can be done quickly and with relatively low effort !
